varats/varats/plot/plots.py

```diff
@@ -0,0 +1,690 @@
+            metavar=self.__metavar,
+            type=self.__type,
+            required=False,
+            help=self.__help
+        )
+
+    def value_or_default(
+        self, default: tp.Optional[OptionType] = None
+    ) -> OptionType:
+        """
+        Retrieve the value for this option.
+
+        The precedence for values is
+        `user provided value > plot-specific default > global default`.
+
+        This function can also be called via the call operator.
+
+        Args:
+            default: plot-specific default value
+
+        Returns:
+            the value for this option
+        """
+        if self.value:
+            return self.value
+        if default:
+            return default
+        return self.default
+
+    def __call__(self, default: tp.Optional[OptionType] = None) -> OptionType:
+        return self.value_or_default(default)
+
+    def __str__(self) -> str:
+        return f"{self.__name}[default={self.__default}, value={self.value}]"
+
+
+class PlotConfig():
+    """
+    Class with parameters that influence a plot's appearance.
+
+    Instances should typically be created with the :func:`from_kwargs` function.
+    """
+
+    def __init__(self, *options: PlotConfigOption[tp.Any]) -> None:
+        self.__options = deepcopy(self._option_decls)
+        for option in options:
+            self.__options[option.name] = option
+
+    _option_decls: tp.Dict[str, PlotConfigOption[tp.Any]] = {
+        decl.name: decl for decl in tp.cast(
+            tp.List[PlotConfigOption[tp.Any]], [
+                PlotConfigOption(
+                    "style",
+                    default="classic",
+                    help_str="Matplotlib style to use."
+                ),
+                PlotConfigOption(
+                    "fig_title",
+                    default="",
+                    help_str="The title of the plot figure."
+                ),
+                PlotConfigOption(
+                    "font_size",
+                    default=10,
+                    help_str="The font size of the plot figure."
+                ),
+                PlotConfigOption(
+                    "width",
+                    default=1500,
+                    help_str="The width of the resulting plot file."
+                ),
+                PlotConfigOption(
+                    "height",
+                    default=1000,
+                    help_str="The height of the resulting plot file."
+                ),
+                PlotConfigOption(
+                    "legend_title",
+                    default="",
+                    help_str="The title of the legend."
+                ),
+                PlotConfigOption(
+                    "legend_size",
+                    default=2,
+                    help_str="The size of the legend."
+                ),
+                PlotConfigOption(
+                    "show_legend",
+                    default=False,
```

Comment:
Maybe change this to True?
```suggestion
                    default=True,
```

Reply:
Hmm, I'd say that it is better if a plot needs to opt in to having a legend (i.e., by using `PlotConfig.show_legend(True)`)

Reply:
Right, this would be better.